### ISSUE TITLE
possible issue on MacOS - using thonny-py5mode

### ISSUE THREAD

**Issue by villares** [hx2A/py5generator#81]
I'm not sure if this is a good place to report (maybe at the thonny-py5mode repo it would be better?), feel free to close,  but I had this strange behavior on a Mac:

<img width="947" alt="Captura de Tela 2022-05-04 às 15 37 01" src="https://user-images.githubusercontent.com/3694604/166803984-4dc86c87-56b2-4929-a568-eb6614039f14.png">

Using `def settings():` [UPDATE: I had previously mixed modes, but still doesn't work]
<img width="743" alt="Captura de Tela 2022-05-04 às 16 12 01" src="https://user-images.githubusercontent.com/3694604/166808813-18a02670-737f-4352-835d-5c01b8a68368.png">

Imported mode worked.
<img width="659" alt="Captura de Tela 2022-05-04 às 15 47 33" src="https://user-images.githubusercontent.com/3694604/166804927-0479e9ae-56a3-4440-85fe-b7b3c20848a9.png">

Thonny 4. beta2 with Python 3.10.4
py5 0.7.2a0

**Comment by hx2A** [comment]
Not sure what is going on here. Can you try putting this in a file:

``` python
def setup():
    size(500, 500)


def draw():
    circle(mouse_x, mouse_y, 10)
```

and then running it at the command line with `run_sketch /path/to/file.py`?

I just tested this on a mac and it worked just fine. It is an old mac, not one of the new Apple Silicon machines. Is your test machine a Mac with an Intel CPU or with Apple Silicon?

**Comment by villares** [comment]
It's an old Mac. I'll see if I can try this tomorrow at the community center!  It's rather baffling!

It's sad that `pip install thonny|`didn't work also.

**Comment by hx2A** [comment]
Stick with it, you'll figure it out!

**Comment by tabreturn** [comment]
I had it running fine on Thonny 3.3.7 with thonny-py5mode version 0.2.5a0

**Comment by villares** [comment]
> I had it running fine on Thonny 3.3.7 (on macOS) with thonny-py5mode version 0.2.5a0

For some reason 3.3.7 was crashing on save here.

If I use 3.3.13 pointing to a different interpreter I can't install thonny-py5mode even if py5 is installed in the 'user interpreter' because the 'IDE interpreter' is the wrong version.

I tried the `sketch_run sketch_path`, it complained about setting JAVA_HOME so I give up for now.

I'm going to close this for now, because as I came back to the machine I thought I had the plug-in installed and it vanished and I can't install it again...
